From 8e2417f122cffeb231dc868dc3743215de4afea0 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Tue, 5 Sep 2023 18:29:25 +0200
Subject: [PATCH 0635/1695] plat-stm32mp2: conf: enable lptimer driver

Add STM32_LPTIMER driver support on stm32mp2 platforms.
It depends on the counter driver. Therefore, enable it if
the STM32_LPTIMER driver is enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9ee47a8a374c852bf77113d401640a170316fc9f plat-stm32mp2: conf: enable lptimer driver]
Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Ia562deb9bca51e48210b06114ebc1a965d61df6d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/326259
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 8112b34c5..b4cc6a613 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -131,6 +131,7 @@ CFG_STM32_I2C ?= y
 CFG_STM32_IAC ?= y
 CFG_STM32_IPCC ?= y
 CFG_STM32_IWDG ?= y
+CFG_STM32_LPTIMER ?= y
 CFG_STM32_OMM ?= y
 CFG_STM32_PWR_IRQ ?= y
 CFG_STM32_RIF ?= y
@@ -234,3 +235,8 @@ CFG_STM32_PANIC_ON_SERC_EVENT ?= n
 else
 CFG_STM32_PANIC_ON_SERC_EVENT ?= y
 endif
+
+# Enable associated counter driver
+ifeq ($(CFG_STM32_LPTIMER),y)
+$(call force,CFG_COUNTER_DRIVER,y)
+endif
-- 
2.47.2

